### DISCUSSION TITLE
SYNC hours

### DISCUSSION THREAD

**Discussion by catch-point** [gopro/labs#134] [Feature Requests]
It would be useful if SYNC would also update hours/date using a store time zone offset.

If `!MZOFF=-5!MSYNC=1` was supported the GoPro date/time clock could be set 5 hours behind UTC time. That would also allow places like St. John's to use `!MOFF=-3.5!MSYNC=1` to be 3.5hours behind UTC (not all timezones are whole hours offset).

**Comment by dnewman-gpsw** [comment]
It does support timezone. e.g.  !MZONE=-420 would allow you to set a new timezone. The timezone is in minutes, fixing the whole hours thing.  It extracts the timezone automatically, using the difference between local time and GPS time upon the first use of !MSYNC=1.  This is saved in the metadata.  Use https://gopro.github.io/labs/control/metadata/ is view the file (header only) metadata from your MP4s.  

In the next HERO11 Labs release (soon), Labs will switch to using the timezone feature in the newer cameras (as TZ is in the menus.)  This will help when moving between timezones.

**Reply by jorisr21** [reply]
Does MZONE work on the HERO10? I would actually be just fine if my camera would simply completely sync to UTC time. A partial sync is rather annoying since the camera looses the clock rather quickly after having no battery. This sync is extremely useful for multi camera recordings. And for decent videos you want those, because (if you ask me) a single camera video is rather boring.

**Reply by dnewman-gpsw** [reply]
Yes it should work, *ZONE=your_offset_in_minutes.  e.g. *ZONE=-480.   Give it a try.

**Reply by jorisr21** [reply]
I tried with !MZONE=0!MSYNC=1 and with !MSYNC=1!MZONE=0 , and it did sync (I changed the minutes manually) , but not to UTC.
I have HERO10 with labs v1.62.70 .

**Reply by dnewman-gpsw** [reply]
Try *ZONE=1 as 0 normally disables the control. It might not work on H10.  Later models have timezones as standard.

**Comment by t0b3** [comment]
FTR: you may sync video dateTime from GPS after recording, i.e. using a shell script
```
#!/bin/bash
for filename in $* ; do
    [ -w $filename ] || return
    utcdate=$(exiftool -m -p /usr/share/exiftool/fmt_files/gpx.fmt -ee $filename | xq -r '.gpx.trk.trkseg.trkpt[-1].time')
    [ -n $utcdate ] || return
    localdate=$(date -d $utcdate '+%Y-%m-%dT%H:%M:%S')
    [ -n $localdate ] || return
    exiftool -overwrite_original -q -FileModifyDate="$localdate" -QuickTime:TrackCreateDate="$localdate" -QuickTime:TrackModifyDate="$localdate" -QuickTime:MediaCreateDate="$localdate" -QuickTime:MediaModifyDate="$localdate" -AllDates="$localdate" $filename
done
```
